helix-view/src/document.rs

```diff
@@ -1093,6 +1098,19 @@ impl Document {
         }
     }
 
+    pub fn version_control_head(&self) -> Option<String> {
+        self.version_control_head
+            .clone()
+            .map(|head| (*head).load().as_ref().as_ref().to_owned())
+    }
```

Comment:
```suggestion
    pub fn version_control_head(&self) -> Option<Arc<Box<str>>> {
        Some(self.version_control_head.as_ref()?.load_full())
    }
```

This is slightly more efficent by avoiding the clone in favor of an `Arc` clone and is a bit shorter.
It's a real shame that `Arc<str>` doesn't work `Option<Arc<str>>` would be much nicer :/ 

We could pull in `slice-dst` and use [`ArcSwap<StrWithHeader<()>>`](https://docs.rs/slice-dst/1.5.1/slice_dst/struct.StrWithHeader.html). I trust both the author and the implementation and it's a very small crate. Not sure if it's worth pulling in for such a smallthing,

Reply:
Nice, that's way better, I like it! I went with an equivalent variant of your suggestion where I'm mapping on the optional instead. It felt slightly cleaner to me.
 
Probably this single use-case will not justify dragging in an extra dependency. I think we achieved a reasonable balance here. Getting rid of `(*head).load().as_ref().as_ref().to_owned()` here is already a huge win.

Great feedback! Thanks! :)